core/ibm_problem.go

```diff
@@ -109,12 +116,21 @@ func (e *IBMProblem) GetCausedBy() Problem {
 // it does not include the error instance the method is called on - that is
 // looked at separately by the "errors" package in functions like "As".
 func (e *IBMProblem) Unwrap() []error {
+	var errs []error
```

Comment:
The changes to this method are the key component of this PR. These changes re-enable compatibility with methods like `errors.Is` by including the previously returned errors in the list of "unwrapped" errors.

The changes to add `Is` implementations to the problem types are purely extra - they allow for comparisons against problems that go beyond being the same instance. They enable equality for the same problem scenario (i.e. same ID) which is what the new types are all about.